tests/main/views/test_direct_award.py

```diff
@@ -548,6 +548,101 @@ def test_award_contract_raises_400_if_project_is_not_locked(self):
         assert res.status_code == 400
 
 
+class TestDirectAwardTellUsAboutContract(TestDirectAwardBase):
+    url = '/buyers/direct-award/g-cloud/projects/1/tell-us-about-contract'
+
+    @pytest.fixture
+    def client(self):
+        self.login_as_buyer()
+        return self.client
+
+    @pytest.fixture
+    def x(self, client):
+        return html.fromstring(client.get(self.url).get_data(as_text=True)).xpath
+
+    @pytest.fixture
+    def data(self):
+        return {
+            'start_date-day': '31',
+            'start_date-month': '12',
+            'start_date-year': '2019',
+            'end_date-day': '1',
+            'end_date-month': '6',
+            'end_date-year': '2021',
+            'buying_organisation': 'Lewisham Council',
+            'value_in_pounds': '100.00'
+        }
+
+    def test_tell_us_about_contract_exists(self, client):
+        assert client.get(self.url).status_code == 200
+
+    def test_tell_us_about_contract_renders(self, x):
+        assert x('boolean(//h1[contains(normalize-space(), "Tell us about your contract")])')
+
+    def test_tell_us_about_contract_form_fields(self, x):
+        assert x('count(//input[@type="text"][substring-after(@name, "start_date")])') == 3
```

Comment:
If `x` is a regular instance function then surely it needs to be called via `self.x`, so I guess there's something that I'm not understanding here if this does actually work.

Either way, I think `x` could be better named, maybe fetch_and_inspect or something, I dunno. I get why you wanted a shorter name, tho, so eh.

Reply:
Instance methods are bound to `__self__` when accessed from an object instance. You can pass around a bound instance method much like any other callable. It's when accessing an instance method through the class that you get an unbound method.

`x` could definitely be better named. How about `xpath`? It's relatively short but also the name of the method, so relatively descriptive.

Reply:
Hmm, but we're actually calling it here - we assert the result of the function, so we must be, no?

xpath is ok but kinda duplicative of the parameter name. It doesn't describe what we're doing, more how we're doing it.